### ISSUE TITLE
AnkiDroid small widget not working

### ISSUE THREAD

**Issue by Mornon** [ankidroid/Anki-Android#5630]
Not sure why, the AnkiDroid small widget stopped working properly: it still shows how many cards are due, but if I tap it AnkiDroid doesn't open. AnkiDroid still works fine, I can open it with the normal icon, it's just AnkiDroid small stopper working, not sure why, it suddendly happened yesterday.

I also tried uninstalling and reinstalling the app, but it still doesn't work.

###### Reproduction Steps

1. Put AnkiDroid small widget on the desktop
2. Tap it to open AnkiDroid

###### Expected Result

AnkiDroid should open.

###### Actual Result

AnkiDroid doesn't open.

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [ x ] character to confirm the points below:*

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

**Comment by Mornon** [comment]
Also the Add note widget is not working.

**Comment by mikehardy** [comment]
That's odd, I have the small widget and it's still working for me, I can't reproduce this. If there was a way for you to get the logs from the device (via logcat) we might be able to see what's going on.

**Comment by Mornon** [comment]
Unfortunately I don't know how to get the logs.

**Comment by Mornon** [comment]
I tried, without success:

- deleting and replacing on desktop the widget, so it doesn't seem to be the widget;
- uninstalling and reinstalling, deleting the AnkiDroid folder and searching for more "AnkiDroid" files (no one found), so it doesn't seem to be the app; it doesn't seem to be the decks either, since by deleting the AnkiDroid folder I had a clean AnkiDroid.

I have no idea what's happening; is there something more I can do, some file I can try deleting before reinstalling that I missed?

**Comment by mikehardy** [comment]
I can't think of anything other than getting the actual logs 

I left a comment elsewhere indicating what I think is the fastest way to do it https://github.com/ankidroid/Anki-Android/issues/5603#issuecomment-550033424

**Comment by Mornon** [comment]
Thanks, that's very useful. I'm now downloading the log: it's some hour it's working, and it's getting kinda big (over 36 MB, right now), will you need all of it?

**Comment by mikehardy** [comment]
oh my! That is quite large. I don't think we will need all of it, but it's hard to say what we will need? Normally the interesting stuff has 'anki' in the line but if the system is failing to launch the app for some reason it may be that it isn't finding the package and the error won't have 'anki' in it at all... If you want you could share it via google drive or dropbox (or anything else) direct to me via github@mikehardy.net

**Comment by Mornon** [comment]
In the log the last rows now are 11-17 followed by local time; does that mean it reached the end of the log and is logging real-time? I'm not sure if I have to stop it manually somehow, or if it will stop itself, since that seems today's date, current time.

**Comment by Mornon** [comment]
I uploaded it on Google Drive, you should receive the link; please let me know if I can do anything else.

**Comment by mikehardy** [comment]
@Mornon I was able to retrieve the thing and scan through it, but it's really odd to me as I don't see anything specifically in there. It does look like the MIUI flavor of android, did you receive a system update recently? You might investigate whether it is being killed by aggressive power saving - https://dontkillmyapp.com/xiaomi?app=AnkiDroid

**Comment by Mornon** [comment]
I didn't have any update, as far as I found last update was in March; the odd thing is, it seems to have stopped working out of the blue, as far as I know nothing has changed.

Anyway I was able to solve it, thanks to your suggestion about the battery saver: it wasn't that, but I started playing with restriction, and turns out this was due to "Start in background" being disabled. No idea why, since I'm sure I didn't change it and it always worked.

Thanks for your help!
